rules.py

```diff
@@ -749,6 +749,38 @@ def check(self):
                     " following CandidateSelections :- %s" % (
                         cand_id, ", ".join(cand_select_ids)))
 
+class ProperBallotSelection(base.BaseRule):
+    """BallotSelections should be correct for that type of contest.
+    
+    Ensure that the BallotSelection elements in a CandidateContest are 
+    CandidateSelections, PartyContests have PartySelections, etc, etc.
+    """
+    
+    con_sel_mapping = {
+        "BallotMeasureContest":"BallotMeasureSelection",
+        "CandidateContest":"CandidateSelection",
+        "PartyContest":"PartySelection",
+        "RetentionContest":"BallotMeasureSelection"
+    }
+    
+    def elements(self):
+        return self.con_sel_mapping.keys()
+    
+    def check(self, element):
+        tag = self.get_element_class(element)
+        selections = []
+        for c in self.con_sel_mapping.keys():
+            selections += self.get_elements_by_class(element,
+                self.con_sel_mapping[c])
+        for selection in selections:
+            selection_tag = self.get_element_class(selection)
+            if (selection_tag != self.con_sel_mapping[tag]):
+                raise base.ElectionError(
+                    "Line %d. The Contest does not contain the right "
```

Comment:
Please add an identifier for the contest (i.e., the objectId) since many of the files we get have all the XML on a single line.